Makefile.kokkos

```diff
@@ -168,7 +168,7 @@ KOKKOS_INTERNAL_COMPILER_CRAY        := $(strip $(shell $(CXX) -craype-verbose 2
 KOKKOS_INTERNAL_COMPILER_NVCC        := $(strip $(shell echo "$(shell export OMPI_CXX=$(OMPI_CXX); export MPICH_CXX=$(MPICH_CXX); $(CXX) --version 2>&1 | grep -c nvcc)>0" | bc))
 KOKKOS_INTERNAL_COMPILER_NVHPC       := $(strip $(shell $(CXX) --version 2>&1 | grep -c "nvc++"))
 KOKKOS_INTERNAL_COMPILER_CLANG       := $(call kokkos_has_string,$(KOKKOS_CXX_VERSION),clang)
-KOKKOS_INTERNAL_COMPILER_CRAY_CLANG  := $(strip $(shell $(CXX) -craype-verbose 2>&1 | grep -c "clang++"))
+KOKKOS_INTERNAL_COMPILER_CRAY_CLANG  := $(strip $(shell $(CXX) -craype-verbose 2>&1 | grep -v "error:" | grep -c "clang++"))
```

Comment:
Why would we filter out "error:"?

Reply:
it is because without that filtering, output will contain multiple lines depending on which compilers to use.

Reply:
Isn't that problematic?

Reply:
I don't think that will be problematic; what I did here is just to fix an error in setting `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG`.

Reply:
`-craype-verbose` is an Cray-Clang-specific option; running this command will cause an error if non-Cray Clang compiler (e.g., Clacc) is used.

Reply:
So would this flag now set to be to true if you use Clacc? Or was it set to true without the filtering out the error?

Reply:
In old Makefile.kokkos, `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG  := $(strip $(shell $(CXX) -craype-verbose 2>&1 | grep -c "clang++"))` sets `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG` to `2` if non-Cray-Clang compilers (e.g., Clacc) are used. However, in later parts of the Makefile, `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG` is used in two cases: 1) true case (`ifeq ($(KOKKOS_INTERNAL_COMPILER_CRAY_CLANG), 1)`) and 2) false case (`ifeq ($(KOKKOS_INTERNAL_COMPILER_CRAY_CLANG), 0)`). However, the false case will not work as intended since the value of `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG` becomes `2` instead of `0` for non-Cray-Clang compilers. This change set `KOKKOS_INTERNAL_COMPILER_CRAY_CLANG` to 0 if non-Cray-Clang compiler is used.